### PR TITLE
Dtj chapter 11

### PR DESCRIPTION
**Pull request by thatguy59**

### CODE REVIEW

File: chapter11/dtj/IslanderDates.pm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    };
+    die "Error getting month name: $@" if $@;
+    return $month[$number];
+}
+
+sub validate_number {
+    my ( $number, $min, $max ) = @_; 
+
+    # The below if statements are all Christmas presents for AlexD
+    if ( !defined $number ){
```
Comment (thatguy59) [inline chapter11/dtj/IslanderDates.pm]: This one is probably covered by `looks_like_number` below

File: chapter11/dtj/IslanderDates.pm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        validate_number( $number, 0, 11 );
+    };
+    die "Error getting month name: $@" if $@;
+    return $month[$number];
+}
+
+sub validate_number {
+    my ( $number, $min, $max ) = @_; 
+
+    # The below if statements are all Christmas presents for AlexD
```
Comment (MarkWignall) [inline chapter11/dtj/IslanderDates.pm]: Lol, he will be so pleased.

File: chapter11/dtj/ex01-tests.pl [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+require 'IslanderDates.pm';
+
+while ( <> ) {
```
Comment (MarkWignall) [inline chapter11/dtj/ex01-tests.pl]: Nice Job writing a test but this test requires input?

File: chapter11/dtj/ex02.pl [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+require 'IslanderDates.pm';
+
+my ( $month, $year, $day ) = (localtime)[4..6];
```
Comment (MarkWignall) [inline chapter11/dtj/ex02.pl]: This line is neat.

File: chapter11/dtj/ex01-tests.pl (hunk 3 above)
Comment (thatguy59) [inline chapter11/dtj/ex01-tests.pl]: Yeah I just wanted an easy way to poke my library.

File: chapter11/dtj/IslanderDates.pm [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+use strict;
+use warnings;
+use Scalar::Util qw ( looks_like_number );
+
+my @day = qw(ark dip wap sen pop sep kir);
+my @month = qw(diz pod bod rod sip wax lin sen kun fiz nap dep);
+sub number_to_day_name {
+    my $number = shift @_;
+    eval {
```
Comment (MarkWignall) [inline chapter11/dtj/IslanderDates.pm]: Do you need these evals?

Comment (thatguy59) [inline chapter11/dtj/IslanderDates.pm]: ~~I wanted the errors to be specific~~

I was debugging a problem and needed the errors to be specific, as I couldn't tell if it was the `number_to_day_name` sub or the `number_to_month_name` sub that was failing (I was calling them both from the same line).

**Pull request merged**